model/src/main/java/org/openremote/model/webhook/Webhook.java

```diff
@@ -0,0 +1,111 @@
+package org.openremote.model.webhook;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+
+public class Webhook {
+
+    public enum Method {
+        GET, POST, PUT, DELETE
+    }
+
+    public enum AuthMethod {
+        NONE, HTTP_BASIC, API_KEY, OAUTH2
+    }
+
+    public static class AuthDetails {
+        public String username;
+        public String password;
+        public String apiKey;
+        public Method method;
+        public String url;
+        public String clientId;
+        public String clientSecret;
+    }
+
+    public static class Header {
```

Comment:
Don't need a class for this can use `Map<String, List<String>>` same as `HTTPAgent` and `HTTPAgentLink` use.